tests/integration-tests/tests/ultraserver/test_gb200/test_gb200/91_nvidia_imex_prolog.sh

```diff
@@ -80,6 +84,47 @@ function get_dna_parameter() {
   jq -r ".cluster.${1}" "${DNA_JSON_FILE}"
 }
 
+function check_imex_needs_reload() {
+  local _expected_ips=$1
+  local _imex_config_file=$2
+  
+  # First check if IMEX service is running
+  if ! systemctl is-active ${IMEX_SERVICE} &>/dev/null; then
+    info "IMEX service is not running, reload needed"
+    return 0  # Need reload
+  fi
+  
+  # Get current IMEX status
+  local imex_status_output
+  if ! imex_status_output=$(timeout 30 /usr/bin/nvidia-imex-ctl -N -j -c "${_imex_config_file}" 2>/dev/null); then
+    info "Failed to get IMEX status, assuming reload needed"
+    return 0  # Need reload
+  fi
+  
+  # Parse JSON to extract current IPs from IMEX status
+  local current_imex_ips
+  if ! current_imex_ips=$(echo "${imex_status_output}" | jq -r '.nodes[].host' 2>/dev/null | sort | tr '\n' ' '); then
```

Comment:
double check it, it does seem aligned with the output schema (it is an array query but the schema is a dictionary): Example:


```
"nodes": {
      "0": {
       "status": "READY",
       "host": "192.168.103.159",
       "connections": {
        "1": {
         "host": "192.168.107.187",
         "status": "CONNECTED",
         "changed": true
        },
```